File: Lesson8_1/Calculator.java
package Lesson8_1;

public class Calculator implements Calc {
    private double operand1;
    private double operand2;
    private String operator;

    public double getOperand1() {
        return operand1;
    }

    public double getOperand2() {
        return operand2;
    }

    public void setOperand1(String operand) {
        this.operand1 = Double.parseDouble(operand);
    }

    public void setOperand2(String operand) {
        this.operand2 = Double.parseDouble(operand);
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public String getResult() {
        Operation operation = operation(operator);
        double result = operation.calculate(operand1, operand2);
        return Double.toString(result);
    }

    private Operation operation(String operator) {
        
        switch (operator) {
            case "+":
                return (op1, op2) -> op1 + op2;
            case "-":
                return (op1, op2) -> op1 - op2;
            case "*":
                return (op1, op2) -> op1 * op2;
            case "/":
                return (op1, op2) -> op2 != 0 ? op1 / op2 : 0;
            default:
                return null;
        }
    }
}

File: Lesson13_2/Calculator.py
import zope.interface
from Calc import Calc


@zope.interface.implementer(Calc)
class Calculator:
    def __init__(self):
        self.__operand1: float
        self.__operand2: float
        self.__operator: str

    def set_operand1(self, operand1):
        self.__operand1 = float(operand1.replace(',', '.'))

    def set_operand2(self, operand2):
        self.__operand2 = float(operand2.replace(',', '.'))

    def set_operator(self, operator):
        self.__operator = operator

    def get_result(self):
        if self.__operator == '+':
            return self.__operand1 + self.__operand2
        elif self.__operator == '-':
            return self.__operand1 - self.__operand2
        elif self.__operator == '*':
            return self.__operand1 * self.__operand2
        elif self.__operator == '/':
            if self.__operand2 == 0:
                print("Division by zero!")
            return self.__operand1 / self.__operand2 if self.__operand2 !=0 else 0
        return 0

